src/controllers/__tests__/authenticationController.test.ts

```diff
@@ -531,6 +607,257 @@ describe("AuthenticationController#acs", () => {
+    ${"with left response"}       | ${Promise.resolve(E.left(new Error("Error on left")))} | ${"Error on left"}
+    ${"with a promise rejection"} | ${Promise.reject(new Error("Error on reject"))}        | ${"Error on reject"}
+  `(
+    "should fail if an error occours deleting the previous CF-assertionRef link on redis $scenario",
+    async ({ delLollipopAssertionRefForUserResponse, errorMessage }) => {
+      const res = mockRes();
+
+      mockGetLollipop.mockResolvedValueOnce(E.right(O.some(anAssertionRef)));
+      mockDelLollipop.mockImplementationOnce(
+        _ => delLollipopAssertionRefForUserResponse
+      );
+
+      mockIsBlockedUser.mockReturnValue(Promise.resolve(E.right(false)));
+      mockGetNewToken
+        .mockReturnValueOnce(mockSessionToken)
+        .mockReturnValueOnce(mockWalletToken)
+        .mockReturnValueOnce(mockMyPortalToken)
+        .mockReturnValueOnce(mockBPDToken)
+        .mockReturnValueOnce(mockZendeskToken)
+        .mockReturnValueOnce(mockFIMSToken)
+        .mockReturnValueOnce(aSessionTrackingId);
+
+      const response = await lollipopActivatedController.acs(validUserPayload);
+      response.apply(res);
+
+      expect(lollipopActivatedController).toBeTruthy();
+      expect(res.status).toHaveBeenCalledWith(500);
+      expect(response).toEqual({
+        apply: expect.any(Function),
+        detail: `Internal server error: ${errorMessage}`,
+        kind: "IResponseErrorInternal"
+      });
+      expect(mockDelLollipop).toBeCalledWith(aFiscalCode);
+      expect(mockActivateLolliPoPKey).not.toBeCalled();
+      expect(mockSetLollipop).not.toBeCalled();
+      expect(mockRevokePreviousAssertionRef).toHaveBeenCalledWith(
+        anAssertionRef
+      );
+
+      expect(mockSet).not.toBeCalled();
+      expect(mockGetProfile).not.toBeCalled();
+      expect(mockCreateProfile).not.toBeCalled();
+    }
+  );
+
+  it(`should fail if an error occours reading the previous CF-assertionRef link on redis`, async () => {
+    const res = mockRes();
+
+    mockGetLollipop.mockResolvedValueOnce(E.left(new Error("Error")));
+
+    mockIsBlockedUser.mockReturnValue(Promise.resolve(E.right(false)));
+    mockGetNewToken
+      .mockReturnValueOnce(mockSessionToken)
+      .mockReturnValueOnce(mockWalletToken)
+      .mockReturnValueOnce(mockMyPortalToken)
+      .mockReturnValueOnce(mockBPDToken)
+      .mockReturnValueOnce(mockZendeskToken)
+      .mockReturnValueOnce(mockFIMSToken)
+      .mockReturnValueOnce(aSessionTrackingId);
+
+    const response = await lollipopActivatedController.acs(validUserPayload);
+    response.apply(res);
+
+    expect(lollipopActivatedController).toBeTruthy();
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(response).toEqual({
+      apply: expect.any(Function),
+      detail:
+        "Internal server error: Error retrieving previous lollipop configuration",
+      kind: "IResponseErrorInternal"
+    });
+    expect(mockDelLollipop).not.toBeCalled();
```

Comment:
we should check that `mockGetLollipop` has been called

Reply:
fixed in ee0346a